src/data/polynomial/degree/erase_lead.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.polynomial.degree.basic
+import data.polynomial.degree.trailing_degree
+
+/-!
+# Erase the leading term of a univariate polynomial
+
+## Definition
+
+* `erase_lead f`: the polynomial `f - leading term of f`
+
+erase_lead serves as the reduction step in an induction, shaving off one monomial from a polynomial.
```

Comment:
```suggestion
* `erase_lead f`: the polynomial `f - leading term of f`

`erase_lead` serves as the reduction step in an induction, shaving off one monomial from a polynomial.
The definition is set up so that it does not mention subtraction in the definition,
and thus works for polynomials over semirings as well as rings.
```

Reply:
I like this addendum!